torch/ao/quantization/fx/prepare.py

```diff
@@ -485,24 +559,16 @@ def _get_arg_target_dtype_as_input_to_node(
     to node `node`
     """
     assert isinstance(arg, Node)
-    is_weight = node_arg_is_weight(node, arg, backend_config)
-    is_bias = node_arg_is_bias(node, arg, backend_config)
-    is_activation = not is_weight and not is_bias
-    if is_activation:
-        input_act_obs_or_fq_ctr = node.meta["target_dtype_info"].get("input_act_obs_or_fq_ctr")
-        qconfig_dtype, _ = _get_dtype_and_is_dynamic(input_act_obs_or_fq_ctr)
-        return qconfig_dtype
-    elif is_weight:
-        if node.target in NON_QUANTIZABLE_WEIGHT_OPS:
-            return None
-        else:
-            weight_obs_or_fq_ctr = node.meta["target_dtype_info"].get("weight_obs_or_fq_ctr", None)
-            qconfig_weight_dtype, _ = _get_dtype_and_is_dynamic(weight_obs_or_fq_ctr)
-            return qconfig_weight_dtype
-    else:
-        bias_obs_or_fq_ctr = node.meta["target_dtype_info"].get("bias_obs_or_fq_ctr", None)
-        qconfig_bias_dtype, _ = _get_dtype_and_is_dynamic(bias_obs_or_fq_ctr)
-        return qconfig_bias_dtype
+    assert "target_dtype_info" in node.meta
+    assert "input_obs_or_fq_ctr_for_args" in node.meta["target_dtype_info"]
+    input_obs_or_fq_ctr_for_args = node.meta["target_dtype_info"]["input_obs_or_fq_ctr_for_args"]
+    arg_to_ctr = {}
+    _populate_arg_to_ctr_for_args(node.args, input_obs_or_fq_ctr_for_args, arg_to_ctr)
+    if "_input_obs_or_fq_ctr_for_kwargs" in node.meta["target_dtype_info"]:
```

Comment:
This will eventually go away for aten graph, right? I am just wondering if we should take a short cut and either do simpler things instead of matching structures, or just assert?

Actually I do see that we are doing simple things, so seems ok to me.

Reply:
yeah this is outdated, I changed it to assert